Feature/add another node

SunjayDK opened the pull request:
Issue #235 

Adds a button to the add node modal that submits but keeps the modal open to add more nodes.

Since this is my first pull request, I would love feedback on code organization. Let me know if there's a better place to put styling or the button addition.

jthrilly commented:
#241

rebeccamadsen commented on src/styles/components/_modal.scss:
```diff
@@ -29,6 +29,11 @@ $component-name: modal;
       overflow-x: hidden;
       overflow-y: scroll;
+
+      .button__container {
```
I don't think this belongs here. We try to use hyphenated bem styles, with a file per component, i.e.:
```
.component__sub-component--variants {
   ...
}
`````
This would mean in your case, I think...
`className="form__button-container"`

// _form.scss
```
.form {
  &__button-container {
     ...
  }
}
```

pfsj commented:
Not sure if these are unique to my system but the form boxes do not clear the previous data after clicking "Add Another" so the form is showing "Name", "Nickname", and "Age" on top of the previous data (This happens in both electron and Chrome).

![screen shot 2017-09-15 at 1 37 12 pm](https://user-images.githubusercontent.com/10948594/30498153-3f4fd674-9a1b-11e7-95f7-f69b74d2f7c3.png)

Also have a couple of more general thoughts for discussion outside the current spec. 

1) I agree that the additional button should take a secondary color to differentiate them. Similarly, the group might want to discuss more clear language for these two actions rather than "Submit" and "Add Another", those names likely would not seem entirely clear to a user.

2) Adding several nodes in a row brings up an issue with knowing who've you've already added. We currently have the side panel to track who you've added in a previous interface and the main panel to show who you've added at the current one but those only work when you return to the main screen after adding each node, which does not happen when using this feature. I don't have an easy solution but perhaps this is something we can discuss in a larger team meeting.

pfsj commented:
So I just did a fresh pull to try and see if this was something local with my version and now I can't replicate the bug I mentioned above (with the text not clearing). 

However, now I'm getting the same functionality of the buttons as Rebecca noticed. That is, the Submit button works correctly but the Add Another button simply alters the functionality of the Submit button when pushed after the Add Another (while also keeping the form open). Was this intended? I believe the spec is supposed to make the Add Another button immediately 'submit' when pressed, without requiring further action.

rebeccamadsen commented:
Hmm, interesting. I did wonder after adding people whether the people I added were really there, but I was typing gibberish, so I chalked that up to me not being a real user with real names. :) We could maybe make a small animation or glow on the nodes after they close the modal, but that again only helps them see what they added once they close the dialog. I could imagine the "Add Another" is most useful when they are initially using the system, adding a bunch at once, so needing to check what they added in previous sessions might not be common? But real users are not just my imaginings...

I did not see the text not clearing bug, so hopefully that was just an issue with your build.

SunjayDK commented on src/containers/Elements/Form.js:
```diff
@@ -41,10 +43,21 @@ class Form extends Component {
+
+    const addAnotherButton = addAnother
+      ? <Button type="button" onClick={continuousSubmit} accessibityLabel="Submit and add another node">Add Another</Button>
+      : null
```
I enabled linting! Thanks for the catch.

SunjayDK commented:
@pfsj 
I restyled the new button and changed the order.
![screen shot 2017-09-19 at 4 47 27 pm](https://user-images.githubusercontent.com/20175029/30617154-69e4c268-9d5a-11e7-9558-d83cb9456103.png)

The issue with text clearing was a bug I saw earlier. I fixed it by adding a form reset, but you may have gotten an earlier version?

I fixed the bug with submit acting like add-another. There's now an awkward animation in chrome when I try it.

SunjayDK commented:
Fixed the submit acting as the add another button bug. Also fixed the two protocol bugs.

Next working on the animation so people can see what nodes are new.

jthrilly commented:
Looks great, Sunjay! Can you rebase against master and just make sure nothing funky happens now we are using a different node version (make sure any snapshots are updated, etc.). Then I think we are good to merge.

@pfsj Your points are all totally valid, and I agree with much of what you said. I think in this case however, our best bet is to ship/test this feature to help us get an idea of how this should ultimately work. 

One idea I just had, for example, is a "peak" feature which lets you look behind the new node form (and see the existing node list) while holding down on the background. This sort of thing would be an experimental branch in itself, either in this release or a future one. For now, we just want to implement the fundamentals so that we can have that discussion.
